### ISSUE TITLE
Skip writing fill-value-only chunks

### ISSUE THREAD

**Issue by forman** [pangeo-data/rechunker#94]
It would be really nice if `rechunker` could exploit the Zarr feature that chunks comprising only fill-values may be missing entirely, given that a `fill_value` is defined for that array. With a new configuration parameter, e.g. `drop_fill_value_chunks`,  `rechunker` could be configured not to write a target chunk at all, given all of its values equal `fill_value`. 

In our `xcube` package we have a dedicated tool called [xcube prune](https://xcube.readthedocs.io/en/latest/cli/xcube_prune.html) for this purpose. But it may be slow for large datasets because it tests and deletes chunks sequentially.

**Comment by d-v-b** [comment]
this new feature added to zarr might be of interest: https://github.com/zarr-developers/zarr-python/pull/738

**Comment by d-v-b** [comment]
I'm not super familiar with the internals of rechunker, but it's possible that `write_empty_chunks` is being specified when an array is created, but ignored when the array is accessed at a later point in time. Unlike stuff like  `compressor` and `chunks`, `write_empty_chunks` is not part of the zarr array metadata that gets written to json; instead, `write_empty_chunks` must be specified each time you create a new instance of `zarr.Array`. If rechunker isn't doing this, then it might explain what you are seeing.

**Comment by d-v-b** [comment]
Glad to hear that was useful, and I don't see any reason why the routines in `creation.py` can't take `write_empty_chunks` as an argument. I can add this to an [ongoing PR](https://github.com/zarr-developers/zarr-python/pull/1051) that addresses this in a different part of the zarr-python codebase.

**Comment by DanielLumb** [comment]
Thanks a lot!
Just note that the modification I needed to make in zarr's `creation.py` was not about `write_empty_chunks`. 
In order to properly work, I needed to pass proper `fill_value`, which `rechunker` passed on to `empty()` in `creation.py`, which does not properly accept `fill_value`, instead enforcing fixed `None` value. That's the line I had to fix:

```
-def empty(shape, **kwargs):
+def empty(shape, fill_value=None, **kwargs):
-    return create(shape=shape, fill_value=None, **kwargs)
+    return create(shape=shape, fill_value=fill_value, **kwargs)
```
